dynatrace/oneagentenrichment/oneagentenrichment.go

```diff
@@ -0,0 +1,168 @@
+// Copyright 2021 Dynatrace LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package oneagentenrichment
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/dynatrace-oss/opentelemetry-metric-go/mint"
+	"go.uber.org/zap"
+)
+
+const indirectionBasename = "dt_metadata_e617c525669e072eebe3d0f08212e8f2"
+
+// OneAgentMetadataEnricher contains the functionality to read and parse
+// OneAgent metadata into key-value pairs.
+type OneAgentMetadataEnricher struct {
```

Comment:
Why is this exported publicly? Do you expect the caller to manually construct without calling `NewOneAgentMetadataEnricher`?